### PR TITLE
go/control: Include node's P2P addresses in control status

### PR DESCRIPTION
**Pull request by peternose**
### Task
Move P2P related status information including node's P2P addresses under P2P section of the control status.

### Test
Note that the data is duplicated if the node is registered (e.g. validators, compute nodes, key managers, but not clients).
```
➜  validator-0 oasis-node control status
{
  "consensus": {
   "p2p": {
      "pub_key": "8EBdmxYwjz9bLreogzNEFuoLPQfci5pRjvHbYHIC/1Q=",
      "peer_id": "21bd3dc64c88cb14ee0d4d5bb8964adc3febdd1b",
      "addresses": [
        "8EBdmxYwjz9bLreogzNEFuoLPQfci5pRjvHbYHIC/1Q=@127.0.0.1:20001"
      ],
      "peers": [
        "78ec9c69741d2a05dd479096d3adcc7c20ab6832@127.0.0.1:20004",
        "a42258a7ecdadfe1f917f737033de17aead2e19a@127.0.0.1:46008",
        "0d57061f4a1b293a8a10f46435eb29055627ee4e@127.0.0.1:46020",
        "ec41b7af0c5522114d5d2d406586e635ceb648a5@127.0.0.1:46036",
        "cfb3f1f210aeb832cd088cd5f6503e5fb820b874@127.0.0.1:46048",
        "02d4b70f0d8afe16a17d298fdaa57dc45b69cf8c@127.0.0.1:46054"
      ]
    }
},
{
  "p2p": {
    "pub_key": "8EBdmxYwjz9bLreogzNEFuoLPQfci5pRjvHbYHIC/1Q=",
    "peer_id": "12D3KooWRzCwZirVmeHqFoA1mruUfTtTQeXrLiz9rzojyAQsjSa7",
    "addresses": [
      "192.168.64.107:20003",
      "127.0.0.1:20003"
    ],
    "num_peers": 6,
    "num_connections": 8
  }
}
```

### CODE REVIEW

File: go/p2p/api/api.go [hunk 1]
```diff
@@ -27,10 +27,28 @@ const (
 	TopicKindTx TopicKind = "tx"
 )
 
+// Status is the P2P status of a node.
+type Status struct {
+	// ID is the public key used for P2P communication.
+	ID signature.PublicKey `json:"id"`
+
+	// Addresses is a list of configured P2P addresses used when registering the node.
+	Addresses []node.Address `json:"addresses"`
```
Comment (kostko) [inline go/p2p/api/api.go:36]: Can we also show the addresses of the native p2p stack (e.g. for libp2p that would be the "peer ID" and for Tendermint P2P this would be the "p2p address")? This is useful for correlating logs as those are the addresses that are logged in corresponding peers.

Comment (peternose) [reply go/p2p/api/api.go:39]: Added `PeerID` to libp2p and moved Tendermint related p2p info under `consensus.p2p` section. See PR's description for the new output.

I wanted to add that earlier for the same reasons but changed my mind as users might be confused with so many ids. For example, Tendermint's p2p status now shows 3 different ids (`id`, `peer_id`, and consensus id inside `addresses`).

File: go/consensus/api/api.go [hunk 2]
```diff
@@ -343,6 +340,24 @@ type Status struct { // nolint: maligned
...
 	IsValidator bool `json:"is_validator"`
+
+	// P2P is the P2P status of the node.
+	P2P *P2PStatus `json:"p2p,omitempty"`
+}
+
+// P2PStatus is the P2P status of a node.
+type P2PStatus struct {
+	// ID is the public key used for consensus P2P communication.
+	ID signature.PublicKey `json:"id"`
```
Comment (kostko) [inline go/consensus/api/api.go:351]: Should we just call this a public key?
```suggestion
	// PubKey is the public key used for consensus P2P communication.
	PubKey signature.PublicKey `json:"pub_key"`
```

File: go/p2p/api/api.go [hunk 3]
```diff
@@ -27,10 +27,31 @@ const (
 	TopicKindTx TopicKind = "tx"
 )
 
+// Status is the P2P status of a node.
+type Status struct {
+	// ID is the public key used for P2P communication.
+	ID signature.PublicKey `json:"id"`
```
Comment (kostko) [inline go/p2p/api/api.go:33]: Should we just call this a public key?
```suggestion
	// PubKey is the public key used for P2P communication.
	PubKey signature.PublicKey `json:"pub_key"`
```

File: go/consensus/api/api.go (hunk 2 above)
Comment (peternose) [reply go/consensus/api/api.go]: We can, but in the registry it would still be `id` :smile:

**Comment by kostko** [comment]
Rebase.

File: go/oasis-node/cmd/node/node_control.go [hunk 4]
```diff
@@ -151,39 +154,34 @@ func (n *Node) GetStatus(ctx context.Context) (*control.Status, error) {
...
 		Registration:    rs,
 		PendingUpgrades: pendingUpgrades,
+		P2P:             p2p,
 	}, nil
 }
 
-// GetIdentity implements control.NodeController.
-func (n *Node) getIdentity() *identity.Identity {
-	return n.Identity
+func (n *Node) getIdentity() control.IdentityStatus {
```
Comment (kostko) [inline go/oasis-node/cmd/node/node_control.go:167]:
```suggestion
func (n *Node) getIdentityStatus() control.IdentityStatus {
```

File: go/control/api/api.go [hunk 5]
```diff
@@ -108,12 +112,6 @@ type IdentityStatus struct {
 	// Node is the node identity public key.
 	Node signature.PublicKey `json:"node"`
 
-	// P2P is the public key used for p2p communication.
-	P2P signature.PublicKey `json:"p2p"`
-
-	// Consensus is the consensus public key.
-	Consensus signature.PublicKey `json:"consensus"`
```
Comment (kostko) [inline go/control/api/api.go:115]: Where is the consensus key reported now? This is unrelated to P2P.

Comment (peternose) [reply go/control/api/api.go:115]: Under `consensus.p2p`, so that it is concise with libp2p.

```
➜  compute-1 oasis-node control status
{
  "p2p": {
    "pub_key": "mrhfjjWQbz6VDNKWCHnLhwWQ5pzUX8Art3AcGhIl+V4=",
  }
},
{
  "consensus": {
    "p2p": {
      "pub_key": "dE9f9JFWYgfSYnZJmfF362516lZsl5OTWhMyk+b7LI8=",
    }
  }
}
```

Comment (kostko) [reply go/control/api/api.go:115]: But this (e.g. what was in `Consensus`) is not the consensus P2P key. So there are two distinct keys:
* The P2P key (used for both libp2p and Tendermint P2P).
* The consensus key (used for signing consensus votes).

So you probably want to add the consensus key back.

Comment (peternose) [reply go/control/api/api.go:115]: I guess I misunderstood that. Will change back.

**Review by kostko — APPROVED** [review]
